### DISCUSSION TITLE
How to add texture to only particular part of the entity?

### DISCUSSION THREAD

**Discussion by kowshik1234** [pokepetter/ursina#610] [Q&A] [answered]
I downloaded an 3D obj from online and when I opened it in UV Editing, it looked as below

<img width="211" alt="image" src="https://github.com/pokepetter/ursina/assets/26191169/d2a8afba-7c75-4557-90d4-8b17db18eef4">

In the above I believe the pink areas is where we can apply the label or texture to, because when I add a label in the UV editing window it comes up correctly.

<img width="665" alt="image" src="https://github.com/pokepetter/ursina/assets/26191169/ed7852e5-11db-4bff-8c73-9c1ea17f6167">

However, when I use the following code in the Urisna, 

```
object = Entity(model='3D_models_path/coke_bottle.obj', scale = 0.2,position=tiles[tile_index].position + (0, 1, 0), texture='coke_texture.jpg')
```
I am getting below result. How do I restrict ursina to apply the label only to the label area and not the entire bottle? 

<img width="182" alt="image" src="https://github.com/pokepetter/ursina/assets/26191169/a5541c88-ef7e-4d58-9d87-f3cc841fcdc1">

Any help is appreciated.

Thanks

**Comment by pokepetter** [comment, marked as answer]
If you use one material and one texture in blender, it will come out correctly. Multiple materials/textures on the same model is not supported since it's bad for performance. It's basically the same as having multiple models.
3D models for games should be made with this in mind. Use atlas textures/trim sheets.
For example, you could reserve one part of the texture for the bottle, one for the label, and one for the cap.
